src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -51,9 +51,20 @@ internal IAggregator[] MapToMetrics(string[] seqKey, object[] seqVal)
             var dt = DateTimeOffset.UtcNow;
 
             // TODO: Need to map each instrument to metrics (based on View API)
-            if (this.instrument.GetType().Name.Contains("Counter"))
+            // TODO: move most of this logic out of hotpath, and to MeterProvider's
+            // InstrumentPublished event, which is once per instrument creation.
+
+            if (this.instrument.GetType() == typeof(Counter<long>)
```

Comment:
I hope this can be modeled as enum + union.

Reply:
not clear what did you mean here.
(this check will be moved out of the hot paths anyway)

Reply:
e.g.

```
struct Metric
{
    internal AggregationType at; // e.g. Add, TakeTheLastValue, etc.
    public PointType pt; // e.g. int8, int16, uint32, double
    [StructLayout(LayoutKind.Explicit)]
    Point
    {
        [FieldOffset(0)]
        byte uint8;
        [FieldOffset(0)]
        ushort uint16;
        ...
    }
}
```

Reply:
got it. Will be part of next rounds :)